DataDog/datadog-operator
# Clean up dependencies for currently reconciled DDA

### What does this PR do?

Applies cleanup method to dependencies associated with the currently reconciled DDA only to prevent other DDA's dependencies from being deleted.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

1. Spin up a DDA, delete it, and ensure that any dependencies managed by the store are deleted. ConfigMaps like `xxx-leader-election` aren't created using the store so those will not be affected.
2. Spin up a DDA with the external metrics feature enabled and check that the corresponding service was created. Update the DDA to disable the external metrics feature and check that the service is deleted.
